src/test/java/com/commercetools/sync/sdk2/cartdiscounts/utils/CartDiscountSyncUtilsTest.java

```diff
@@ -0,0 +1,342 @@
+import com.commercetools.api.models.cart_discount.CartDiscountSetValidFromAndUntilActionBuilder;
+import com.commercetools.api.models.cart_discount.CartDiscountTarget;
+import com.commercetools.api.models.cart_discount.CartDiscountUpdateAction;
+import com.commercetools.api.models.cart_discount.CartDiscountValue;
+import com.commercetools.api.models.cart_discount.CartDiscountValueAbsoluteDraftBuilder;
+import com.commercetools.api.models.cart_discount.CartDiscountValueDraft;
+import com.commercetools.api.models.cart_discount.CartDiscountValueRelativeBuilder;
+import com.commercetools.api.models.cart_discount.CartDiscountValueRelativeDraftBuilder;
+import com.commercetools.api.models.cart_discount.StackingMode;
+import com.commercetools.api.models.common.CentPrecisionMoneyBuilder;
+import com.commercetools.api.models.common.LocalizedString;
+import com.commercetools.api.models.type.CustomFields;
+import com.commercetools.api.models.type.CustomFieldsDraft;
+import com.commercetools.api.models.type.CustomFieldsDraftBuilder;
+import com.commercetools.api.models.type.FieldContainer;
+import com.commercetools.api.models.type.TypeReferenceBuilder;
+import com.commercetools.sync.sdk2.cartdiscounts.CartDiscountSyncOptions;
+import com.commercetools.sync.sdk2.cartdiscounts.CartDiscountSyncOptionsBuilder;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import java.time.ZonedDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class CartDiscountSyncUtilsTest {
+
+  private static final ProjectApiRoot CTP_CLIENT = mock(ProjectApiRoot.class);
+
+  private static final String KEY = "key";
+  private static final LocalizedString NAME = LocalizedString.ofEnglish("name");
+  private static final LocalizedString DESC = LocalizedString.ofEnglish("discount- get 10 percent");
+  private static final String PREDICATE =
+      "lineItemExists(sku = \"0123456789\" or sku = \"0246891213\") = true";
+  private static final CartDiscountValue VALUE =
+      CartDiscountValueRelativeBuilder.of().permyriad(1000L).build();
+
+  private static final CartDiscountValueDraft VALUE_DRAFT =
+      CartDiscountValueRelativeDraftBuilder.of().permyriad(1000L).build();
+  private static final CartDiscountTarget TARGET =
+      CartDiscountLineItemsTargetBuilder.of().predicate("1=1").build();
+  private static final ZonedDateTime JANUARY_FROM = ZonedDateTime.parse("2019-01-01T00:00:00.000Z");
+  private static final ZonedDateTime JANUARY_UNTIL =
+      ZonedDateTime.parse("2019-01-31T00:00:00.000Z");
+  private static final String SORT_ORDER = "0.1";
+  private static final Boolean IS_ACTIVE = false;
+  private static final Boolean IS_REQUIRE_DISC_CODE = false;
+  private static final StackingMode STACKING_MODE = StackingMode.STACKING;
```

Comment:
Most of these fields are used in @BeforeEach only. You could move there.

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/1016/commits/78909c7288f5d5e9cb250af14a4f1e16c6e0a5be